utils/controller_debug.py

```diff
@@ -58,6 +66,84 @@
 # It can be toggled with the 'debug' command
 showSerial = False
 
+# Copied from network_protocol.proto.
+# TODO: Import the proto instead!
+VentMode_OFF = 0
+VentMode_PRESSURE_CONTROL = 1
+VentMode_PRESSURE_ASSIST = 2
+
+
+@dataclasses.dataclass
+class Preset:
+    """A named list of DebugVars + values."""
+
+    name: str
+    desc: str
+    vars: Dict[str, Union[int, float]]
+
+    def ShortDesc(self):
+        return f"{self.name} - {self.vars}"
+
+    def LongDesc(self):
+        return (
+            self.desc
+            + "\n"
+            + "\n".join(f"  - {var} = {val}" for var, val in self.vars.items())
+        )
+
+
+def CoventPCPreset(
+    test_num,
+    intended_tv,
+    lung_compliance,
+    lung_resistance,
+    rr,
+    inspiratory_time,
+    delta_inspiratory_pressure,
+    fio2,
+    bap,
+):
+    """Constructs a Preset object for a particular covent pressure-control test.
+
+    Copied from CoVent-19 Ventilator Testing Procedure, table 201.105:
+    https://drive.google.com/file/d/1FJOs6pdwHqV-Ygm5gMwIRBAmqH6Xxby8
+    """
+    desc = f"""\
+CoVent-19 pressure-control test #{test_num}
+
+If you have a calibrated test lung, configure it as follows:
+
+ - Compliance: {lung_compliance} ml/hPa +/- 10%
+ - Linear resistance: {lung_resistance} hPa/l/s +/- 10%
+
+Intended TV: {intended_tv} ml
+"""
+
+    sec_per_breath = 60 / rr
+    ie_ratio = inspiratory_time / sec_per_breath
+    vars = {
+        "gui_mode": VentMode_PRESSURE_CONTROL,
+        "gui_bpm": rr,
+        "gui_ie_ratio": round(ie_ratio, 2),
+        "gui_pip": bap + delta_inspiratory_pressure,
+        "gui_peep": bap,
+        # TODO: FiO2 not currently supported.
+    }
+    return Preset(f"covent_pc_{test_num}", desc, vars)
+
+
+# DebugVar presets recognized by the `preset` command.
+PRESETS = [
+    CoventPCPreset(1, 500, 50, 5, 20, 1, 10, 30, 5),
```

Comment:
Perhaps use named arguments here?

Reply:
Just tried, and even with cryptically-short names, I can't get it to fit on one line.

```
    CoventPC(n=1, tv=500, cpl=50, res=5, rr=20, itime=1, ip=10, fio2=30, bap=15), # 81 chars
```

If we split it up into multiple lines I'm afraid it's going to be easier to have a transcription error.